### DISCUSSION TITLE
Help using tree-sitter in my VSCode Extension

### DISCUSSION THREAD

**Discussion by MyPingO** [tree-sitter/tree-sitter#3117] [Q&A] [answered]
I am running into dozens of issues using tree-sitter library in my JavaScript Extension for VSCode. Can I get someone to please explain how to properly install it.

After many issues with version mismatches with Electron, when I try to initialize the parser, my extension just crashes without and logs.

```
const Parser = require("tree-sitter"); // <---- Crashes here
const javaScript = require("tree-sitter-javascript");
const parser = new Parser();
```

**Comment by sogaiu** [comment]
Not specific to your crash, but may be there's some project mentioned [here](https://github.com/tree-sitter/tree-sitter/discussions/2903#discussioncomment-8358181) that might have a working setup that can be compared with.

**Reply by MyPingO** [reply]
Hi @sogaiu I did look at these repositories but I am still unable to understand what the problem is. I tried using `web-tree-sitter` but that comes with its own share of problems.
It seems like tree-sitter has many issues working well with VSCode. Know of any alternatives?

**Reply by sogaiu** [reply]
I haven't recently been working with VSCode extensions, so I'm not sure what the state of things is (^^;

I think at least @pokey (the cursorless dev -- related to VSCode extension) and I believe @savetheclocktower (pulsar dev -- because of know-how in using web-tree-sitter in an editor context) might know a lot better about the state of things.

They might be reachable via the discord server (see link at the top of [here](https://github.com/tree-sitter/tree-sitter/discussions) if not known already).

**Reply by pokey** [reply, marked as answer]
@MyPingO we have a working setup in https://github.com/cursorless-dev/vscode-parse-tree/; feel free to fork that

**Reply by MyPingO** [reply]
Yes, it seems you use `web-tree-sitter` which is what I ended up using due to infinite conflicts with VSCode/Electron.
I should have closed the issue.

**Reply by pokey** [reply]
Oh yeah we tried to use native tree-sitter and bailed pretty quick as it was a nightmare to get that working with vscode. Maintaining a native VSCode extension is a serious undertaking 😅
